### PR TITLE
[0.6] Make sure to update the name in the mutator

### PR DESCRIPTION
**Pull request by JonCrowther**
Fixes an issue with https://github.com/rancher/webhook/pull/532 where the name wasn't being updated

### CODE REVIEW

**Review by MbolotSuse — COMMENTED** [review]
Idea looks ok to me (assuming that if you set both name and generate name k8s won't just re-make the name based on generate name, though I would test that). I would definitely add a test for this overall logic if we can.

**Review by nflynt — APPROVED** [review]

File: pkg/resources/management.cattle.io/v3/project/mutator.go [hunk 1]
```diff
@@ -126,8 +126,9 @@ func (m *Mutator) createProjectNamespace(project *v3.Project) (*v3.Project, erro
 	// for the backing namespace, we need to generate it ourselves
 	if project.Name == "" {
 		// If err is nil, (meaning "project exists", see below) we need to repeat the generation process to find a project name and backing namespace that isn't taken
+		var newName string
 		for err == nil {
-			newName := names.SimpleNameGenerator.GenerateName(project.GenerateName)
+			newName = names.SimpleNameGenerator.GenerateName(project.GenerateName)
```
Comment (pmatseykanets) [inline pkg/resources/management.cattle.io/v3/project/mutator.go:131]: I'd probably just assign it directly instead and use it in code that follows
```suggestion
			newProject.Name = names.SimpleNameGenerator.GenerateName(project.GenerateName)
```

**Review by pmatseykanets — DISMISSED** [review]

**Comment by JonCrowther** [comment]
> Idea looks ok to me (assuming that if you set both name and generate name k8s won't just re-make the name based on generate name, though I would test that). I would definitely add a test for this overall logic if we can.

I have tested the generateName vs name, and as long as name is populated, k8s won't use generate name to overwrite it

**Review by pmatseykanets — APPROVED** [review]

**Review by nflynt — APPROVED** [review]
:+1: Tests look like they'd cover the documented business requirements for the generative name prefix.

**Pull request merged**